paris-material/src/main/java/com/airbnb/paris/material/proxies/MaterialButtonProxy.kt

```diff
@@ -0,0 +1,161 @@
+    @Attr(R2.styleable.Paris_MaterialButton_icon)
+    fun setIcon(icon: Drawable){
+        view.icon = icon
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_iconSize)
+    fun setIconSize(@Px size: Int){
+        view.iconSize = size
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_iconGravity)
+    fun setIconGravity(gravity: Int){
+        view.iconGravity = gravity
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_iconPadding)
+    fun setIconPadding(@Px padding: Int){
+        view.iconPadding = padding
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_iconTint)
+    fun setIconTintList(tintList:ColorStateList?){
+        view.iconTint = tintList
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_iconTintMode)
+    fun setIconTintModeMode(mode: Int){
+        val tintMode = ViewUtils.parseTintMode(mode, PorterDuff.Mode.SRC_IN)
+        view.iconTintMode = tintMode
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_rippleColor)
+    fun setRippleColor(ripple:ColorStateList?){
+        view.rippleColor = ripple
+
+        Log.d("MaterialProxy", "ripple: $ripple")
+
+    }
+
+    //todo add shapeAppearance & shapeAppearanceOverlay
+
+    @Attr(R2.styleable.Paris_MaterialButton_strokeColor)
+    fun setStrokeColor(strokeColor: ColorStateList?){
+        view.strokeColor = strokeColor
+
+        Log.d("MaterialProxy", "strokeColor: $strokeColor")
+    }
+
+    @Attr(R2.styleable.Paris_MaterialButton_strokeWidth)
+    fun setStrokeWidth(@Px width: Int){
+        view.strokeWidth = width
+
+        Log.d("MaterialProxy", "strokeWidth: $width")
+    }
+
+    @AfterStyle
+    fun afterStyle(@Suppress("UNUSED_PARAMETER") style: Style?) {
+        /*
+        * Material component has default textAppearance.
+        * But if we do not pass any textAppearance then
+        * android default textAppearance is set. To keep the
+        * material theme textAppearance I have checked
+        * whether it is the default android textAppearance.
+        * If it's default android textAppearance then I have
+        * assumed that user did not set the textAppearance
+        * */
+        if(mTextAppearance == defaultTextAppearance){
+
+            if(android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M){
+                view.setTextAppearance(R.style.TextAppearance_MaterialComponents_Button)
+            } else{
+                view.setTextAppearance(view.context, R.style.TextAppearance_MaterialComponents_Button)
+            }
+
+        } else{
+
+            if(android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M){
+                view.setTextAppearance(mTextAppearance)
+            } else{
+                view.setTextAppearance(view.context, mTextAppearance)
+            }
+
+        }
+
+        if(mColors == null){
+            val typedValue = TypedValue()
+            view.context.theme.resolveAttribute(R.attr.colorOnPrimary, typedValue, true)
+            view.setTextColor(typedValue.data)
```

Comment:
How is using `colorOnPrimary` implemented in `MaterialButton`? This may be unnecessary.